src/test/java/org/skife/jdbi/v2/BeanMapperTest.java

```diff
@@ -0,0 +1,270 @@
+ * limitations under the License.
+ */
+
+package org.skife.jdbi.v2;
+
+import org.easymock.EasyMockRunner;
+import org.easymock.Mock;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.skife.jdbi.v2.util.BigDecimalColumnMapper;
+import org.skife.jdbi.v2.util.IntegerColumnMapper;
+import org.skife.jdbi.v2.util.LongColumnMapper;
+import org.skife.jdbi.v2.util.StringColumnMapper;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+
+@RunWith(EasyMockRunner.class)
+public class BeanMapperTest {
+
+    @Mock
+    ResultSet resultSet;
+
+    @Mock
+    ResultSetMetaData resultSetMetaData;
+
+    @Mock
+    StatementContext ctx;
+
+    BeanMapper<SampleBean> mapper = new BeanMapper<SampleBean>(SampleBean.class);
+
+    @Test
+    public void shouldSetValueOnPublicSetter() throws Exception {
+        expect(resultSetMetaData.getColumnCount()).andReturn(1).anyTimes();
+        expect(resultSetMetaData.getColumnLabel(1)).andReturn("longField");
+        replay(resultSetMetaData);
+
+        expect(ctx.columnMapperFor(Long.class)).andReturn(LongColumnMapper.WRAPPER);
+        replay(ctx);
+
+        expect(resultSet.getMetaData()).andReturn(resultSetMetaData);
+        Long aLongVal = 100l;
+        expect(resultSet.getLong(1)).andReturn(aLongVal);
+        expect(resultSet.wasNull()).andReturn(false);
+        replay(resultSet);
+
+        SampleBean sampleBean = mapper.map(0, resultSet, ctx);
+
+        assertSame(aLongVal, sampleBean.getLongField());
+    }
+
+    @Test
+    public void shouldHandleEmptyResult() throws Exception {
+        expect(resultSetMetaData.getColumnCount()).andReturn(0);
+        replay(resultSetMetaData);
+
+        expect(resultSet.getMetaData()).andReturn(resultSetMetaData);
+        replay(resultSet);
+
+        SampleBean sampleBean = mapper.map(0, resultSet, ctx);
+
+        assertNotNull(sampleBean);
+    }
+
+    @Test
+    public void shouldBeCaseInSensitiveOfColumnAndPropertyNames() throws Exception {
+        expect(resultSetMetaData.getColumnCount()).andReturn(1).anyTimes();
+        expect(resultSetMetaData.getColumnLabel(1)).andReturn("LoNgfielD");
+        replay(resultSetMetaData);
+
+        expect(ctx.columnMapperFor(Long.class)).andReturn(LongColumnMapper.WRAPPER);
+        replay(ctx);
+
+        expect(resultSet.getMetaData()).andReturn(resultSetMetaData);
+        Long aLongVal = 100l;
+        expect(resultSet.getLong(1)).andReturn(aLongVal);
+        expect(resultSet.wasNull()).andReturn(false);
+        replay(resultSet);
+        SampleBean sampleBean = mapper.map(0, resultSet, ctx);
+
+        assertSame(aLongVal, sampleBean.getLongField());
```

Comment:
`assertSame` here unintentionally tests the `Long` cache, if I'm not mistaken... is `assertEquals` better?